harmonica/meshes.py

```diff
@@ -0,0 +1,212 @@
+
+    >>> region = [-10.1, 10.1, -10.1, 10.1]
+    >>> layer = tesseroids_layer(region, spacing=0.5)
+    >>> print(layer.region)
+    (-10.1, 10.1, -10.1, 10.1)
+    >>> print("({:.3f}, {:.3f})".format(*layer.spacing))
+    (0.505, 0.505)
+
+    Or the region:
+
+    >>> layer = tesseroids_layer(region, spacing=0.5, adjust="region")
+    >>> print(layer.region)
+    (-10.1, 9.9, -10.1, 9.9)
+    >>> print(layer.spacing)
+    (0.5, 0.5)
+
+    Create a layer with top, bottom and density
+
+    >>> from harmonica import get_ellipsoid
+    >>> ellipsoid = get_ellipsoid()
+    >>> top = ellipsoid.mean_radius
+    >>> bottom = top - 1e3
+    >>> density = 2670
+    >>> region = [-70.1, -60.2, -45.2, -33.1]
+    >>> layer = tesseroids_layer(
+    ...     region, spacing=0.5, top=top, bottom=bottom, density=density
+    ... )
+    >>> print("({:.3f}, {:.3f})".format(*layer.spacing))
+    (0.504, 0.495)
+    >>> print(layer.shape)
+    (25, 21)
+    >>> print(layer.data_vars)
+    Data variables:
+        top      (latitude, longitude) float64 6.371e+06 6.371e+06 ... 6.371e+06
+        bottom   (latitude, longitude) float64 6.37e+06 6.37e+06 ... 6.37e+06
+        density  (latitude, longitude) float64 2.67e+03 2.67e+03 ... 2.67e+03
+
+    Pass top and bottom as ellipsoidal heights (geodetic coordinate system)
+
+    >>> bottom, top = -1000, 1000
+    >>> region = [-12, 12, -20, 20]
+    >>> layer = tesseroids_layer(
+    ...     region, spacing=0.5, top=top, bottom=bottom, coordinates="geodetic"
+    ... )
+    >>> print(layer.data_vars)
+    Data variables:
+        top      (latitude, longitude) float64 6.377e+06 6.377e+06 ... 6.377e+06
+        bottom   (latitude, longitude) float64 6.375e+06 6.375e+06 ... 6.375e+06
+        density  (latitude, longitude) float64 0.0 0.0 0.0 0.0 ... 0.0 0.0 0.0 0.0
+
+    Create a layer where region coordinates are the boundaries of extreme tesseroids
+
+    >>> region = [-1, 1, -1, 1]
+    >>> layer = tesseroids_layer(region, spacing=0.2, region_centers=False)
+    >>> print(layer.coords)
+    Coordinates:
+      * longitude  (longitude) float64 -0.9 -0.7 -0.5 -0.3 -0.1 0.1 0.3 0.5 0.7 0.9
+      * latitude   (latitude) float64 -0.9 -0.7 -0.5 -0.3 -0.1 0.1 0.3 0.5 0.7 0.9
+
+    """
+    check_region(region)
+    if shape is not None and spacing is not None:
+        raise ValueError("Both shape and spacing provided. Only one is allowed.")
+    if shape is None and spacing is None:
+        raise ValueError("Either a shape or a spacing must be provided.")
+    if spacing is not None:
+        shape, region = spacing_to_shape(region, spacing, adjust)
+    if coordinates not in ("spherical", "geodetic"):
+        raise ValueError(
+            "Invalid geographic coordinate system '{}'.".format(coordinates)
+            + " Should be 'spherical' or 'geodetic'."
+        )
+    longitude = np.linspace(region[0], region[1], shape[1])
+    latitude = np.linspace(region[2], region[3], shape[0])
```

Comment:
Is there a reason for repeating all of these lines instead of using `verde.grid_coordinates`?

Reply:
I wanted to take advantage of xarray coordinates: we don't need to make `np.meshgrid(longitude, latitude)`, so I though we can save a little bit of memory by omitting it.

Nevertheless I was just drafting, so we can still think how to reduce code repetition.